### ISSUE TITLE
Support TRAPI 1.2 (from the current 1.1)

### ISSUE THREAD

**Issue by newgene** [biothings/biothings_explorer#248]
TRAPI v1.2 CHANGE LOGS is here: https://github.com/NCATSTranslator/ReasonerAPI/blob/v1.2/ChangeLog.md

- [x] /predicates endpoint is REMOVED https://github.com/NCATSTranslator/ReasonerAPI/pull/261/files
      - can be done quickly, but even leave it there is not a problem.
      - done in #294 

- [x] Removed all 'relation' properties in Edge, QEdge, MetaEdge

- [ ] /meta_knowledge_graph allows optional attributes[] property (encouraged but not required) https://github.com/NCATSTranslator/ReasonerAPI/pull/259/files

- [x] Add support for workflow and operations v1.0.0 (encouraged but not yet required) https://github.com/NCATSTranslator/ReasonerAPI/pull/262/files

- [x] #364
      - implemented and deployed in dev instance for testing now.
     
- [x] Add result.score (highly encouraged but not required) https://github.com/NCATSTranslator/ReasonerAPI/pull/247/files

- [x] /meta_knowledge_graph should only include specific relationships, not ancestor-inferred relationships (policy clarification)

- [ ] Add one level of subattributes to Attribute https://github.com/NCATSTranslator/ReasonerAPI/pull/268/files
       This is targeted to some specific use-cases, a minor feature we don't have to add right now.

- [ ] Add attributes[] to EdgeBinding to complement Edge and Node https://github.com/NCATSTranslator/ReasonerAPI/pull/269/files
       This is also another minor enhancement we don't have to worry for now, we can add it easily when we have specific use case.

After we are done the 1.2 support, label the `info.x-trapi.version` as `1.21` in SmartAPI registry

**Comment by colleenXu** [comment]
I'm noticing that meta_knowledge_graph is different for APIs with TRAPI v1.2 already implemented....and we may need to update our parser to be able to handle this. 

The objects within edges have an attributes object within them now, as well as the subject/object/predicate fields that we want. 

Example is these [COHD](https://covid.cohd.io/api/meta_knowledge_graph) [endpoints](https://trapi-dev.cohd.io/api/meta_knowledge_graph) that we currently don't ingest yet.

**Comment by colleenXu** [comment]
Note the changes might now be here in the release log: https://github.com/NCATSTranslator/ReasonerAPI/releases/tag/v1.2.0

**Comment by colleenXu** [comment]
once this issue is complete, we can update the smartapi registration which involves what chunlei said above and edits like removing the /predicates endpoint if necessary...

> After we are done the 1.2 support, label the info.x-trapi.version as 1.21 in SmartAPI registry

**Comment by colleenXu** [comment]
@newgene Should we label BTE as 1.2.1 or 1.2.0? 1.2.0 is the latest formal release?

FYI, these issues are addressed + closed. They had to do with BTE ingesting TRAPI v1.2 correctly: #274 #269

**Comment by newgene** [comment]
If I remember correctly it should be `1.2.1` @colleenXu

**Comment by colleenXu** [comment]
okay, it's been changed to [1.2.1](https://github.com/NCATS-Tangerine/translator-api-registry/blob/7ca3a79aaaf977ba75f214f4983fcedcfe038c41/biothings_explorer/smartapi.yaml#L12)

---

This is from the Translator ReasonerAPI call notes for 9/16/21 from "last week", bold emphasis is mine:

> Note the switch from nullable:true to default: [ ] #292
> https://github.com/NCATSTranslator/ReasonerAPI/pull/292/files
> DECISION: Seems fine. Request approvals. **Release as 1.2.1 along with PR 219**

Investigating, it looks like that PR 292 refers to another PR, [286](https://github.com/NCATSTranslator/ReasonerAPI/pull/286/files). Those two lines for default: [] are already present in BTE's spec [here](https://github.com/NCATS-Tangerine/translator-api-registry/blob/4cac5c11a5e9fc826dcaf12f1d3a97845a1cd41f/biothings_explorer/smartapi.yaml#L685) and [here](https://github.com/NCATS-Tangerine/translator-api-registry/blob/4cac5c11a5e9fc826dcaf12f1d3a97845a1cd41f/biothings_explorer/smartapi.yaml#L726).

**Comment by colleenXu** [comment]
note though that this view does not update right away even after I try to refresh the api...http://smart-api.info/registry?q=dc91716f44207d2e1287c727f281d339

**Comment by colleenXu** [comment]
Related to #301

**Comment by colleenXu** [comment]
The other points seem to be optional. I suggest potentially closing this issue for now, or only keeping it open as documentation of what else we could do and still be TRAPI v1.2 compliant.

The other points that are still open right now:
*  /meta_knowledge_graph allows optional attributes[] property (encouraged but not required) https://github.com/NCATSTranslator/ReasonerAPI/pull/259/files
* Add one level of subattributes to Attribute https://github.com/NCATSTranslator/ReasonerAPI/pull/268/files
* Add attributes[] to EdgeBinding to complement Edge and Node https://github.com/NCATSTranslator/ReasonerAPI/pull/269/files

**Comment by colleenXu** [comment]
Note that change log is now here: https://github.com/NCATSTranslator/ReasonerAPI/blob/v1.2.0/ChangeLog.md
